addons/cppcheckdata.py

```diff
@@ -1485,7 +1485,7 @@ def reportError(location, severity, message, addon, errorId, extra=''):
                 'linenr': location.linenr,
                 'column': location.column,
                 'severity': severity,
-                'message': message,
+                'message': message + ' (' + extra + ')',
```

Comment:
this is unfortunate there are many addons that doesn't provide an "extra" argument

Instead of just appending it to the message. My feeling is that we should control how its written by cppcheck through the normal cppcheck option --template instead. It can be added in paratheses in the default format. The information could be added in the xml output. And there can be a separate column in the GUI where the "Advisory" is shown.

Reply:
I am not sure I understand what you would like me to do here. 

Isn't the `--template` option for the standard output ? In this use case, we use the `--xml` argument to export the results to an xml file.

```bash
cppcheck --addon=misra.json --xml main.cpp 2>> cppcheck_out.xml
cppcheck-codequality --input-file=cppcheck_out.xml --output-file=cppcheck.json
```

Would it be acceptable to append the `extra` value to the message only if `extra` is defined ? I.e. copy/pasting and adapting lines 1497-1498 to line 1492 ? 
```python
        if len(extra) > 0:
            msg["message"] += ' (' + extra + ')'
```

Reply:
I don't have a strong opinion about the misra output. It might make sense to do this for misra output. But do not change the output for all the other addons.